workflow/libero_flow_poc/libero_flow/flow/workflow/tests/test_api/test_activity_api.py

```diff
@@ -0,0 +1,17 @@
+import pytest
+
+from workflow.models import Activity
+
+
+@pytest.mark.django_db
+def test_can_create_activity_via_post(admin_client, valid_activity_data):
+    response = admin_client.post('/workflows/api/v1/activities/', data=valid_activity_data)
```

Comment:
what is the use case for creating an activity via API outside of a larger workflow? Would it be able to run on its own?

Reply:
The `scheduler` uses that to create each initial activity state and the `Activity` status is
updated using a `PATCH` on the same endpoint but providing its id.

This means I did not need to give the other processes database access, could be run anywhere and just use the API.

Reply:
I'm all for exposing APIs to the other services rather than giving access to the internals like a database (or a bucket or whatever resource is being used), although in the case of these containers they all come from the same source code so I would expect them to run in the same deployment, maybe not on the same node or Kubernetes pod but definitely inside the same VPC. So giving access to the database should not be a big deal for these processes, while it may be useful to go through the API for extensions (we don't know yet what those would do or how they would be implemented).

Reply:
Agreed, if need be, with the current project structure you could easily give them an interface to the database. Though as you have mentioned, currently I can do other things when the API requests are processed which is handy which you may not know about if the tables were just updated directly.